trino/base/hive-metastore-entrypoint.yaml

```diff
@@ -0,0 +1,120 @@
+      PEM_FILE=$1
+      PASSWORD=$2
+      KEYSTORE=$3
+      # number of certs in the PEM file
+      CERTS=$(grep 'END CERTIFICATE' $PEM_FILE| wc -l)
+
+      # For every cert in the PEM file, extract it and import into the JKS keystore
+      # awk command: step 1, if line is in the desired cert, print the line
+      #              step 2, increment counter when last line of cert is found
+      for N in $(seq 0 $(($CERTS - 1))); do
+        ALIAS="${PEM_FILE%.*}-$N"
+        cat $PEM_FILE |
+          awk "n==$N { print }; /END CERTIFICATE/ { n++ }" |
+          keytool -noprompt -import -trustcacerts \
+                  -alias $ALIAS -keystore $KEYSTORE -storepass $PASSWORD
+      done
+    }
+    set -e
+
+    # if the s3-compatible ca bundle is mounted in, add to the root Java truststore.
+    if [ -a /s3-compatible-ca/ca-bundle.crt ]; then
+      echo "Adding /s3-compatible-ca/ca-bundle.crt to $JAVA_HOME/lib/security/cacerts"
+      importCert /s3-compatible-ca/ca-bundle.crt changeit $JAVA_HOME/lib/security/cacerts
+    fi
+    # always add the openshift service-ca.crt if it exists
+    if [ -a /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt ]; then
+      echo "Adding /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt to $JAVA_HOME/lib/security/cacerts"
+      importCert /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt changeit $JAVA_HOME/lib/security/cacerts
+    fi
+
+    # add UID to /etc/passwd if missing
+    if ! whoami &> /dev/null; then
+        if test -w /etc/passwd || stat -c "%a" /etc/passwd | grep -qE '.[267].'; then
+            echo "Adding user ${USER_NAME:-hadoop} with current UID $(id -u) to /etc/passwd"
+            # Remove existing entry with user first.
+            # cannot use sed -i because we do not have permission to write new
+            # files into /etc
+            sed  "/${USER_NAME:-hadoop}:x/d" /etc/passwd > /tmp/passwd
+            # add our user with our current user ID into passwd
+            echo "${USER_NAME:-hadoop}:x:$(id -u):0:${USER_NAME:-hadoop} user:${HOME}:/sbin/nologin" >> /tmp/passwd
+            # overwrite existing contents with new contents (cannot replace the
+            # file due to permissions)
+            cat /tmp/passwd > /etc/passwd
+            rm /tmp/passwd
+        fi
+    fi
+
+    # symlink our configuration files to the correct location
+    if [ -f /hadoop-config/core-site.xml ]; then
+      cat /hadoop-config/core-site.xml | sed "s#XXX_S3ENDPOINT_XXX#${S3_BUCKET_NAME}/${S3_DATA_DIR}#" > /etc/hadoop/core-site.xml
+    else
+      echo "core-site.xml doesnt exist, skipping"
+    fi
+    if [ -f /hadoop-config/hdfs-site.xml ]; then
+      ln -s -f /hadoop-config/hdfs-site.xml /etc/hadoop/hdfs-site.xml
+    else
+      echo "hdfs-site.xml doesnt exist, skipping symlink"
+    fi
+
+    # insert S3 bucket URI from env vars
+    cat /hive-config/hive-site.xml | sed \
+      -e "s#XXX_S3LOCATION_XXX#${S3_ENDPOINT_URL_PREFIX}#" \
```

Comment:
This appears to be a bug since you are replacing `XXX_S3LOCATION_XXX` twice with 2 different values.
I think you wanted to set this to 
```
-e "s#XXX_S3ENDPOINT_URL_PREFIX_XXX#${S3_ENDPOINT_URL_PREFIX}#" \
```

Reply:
yes you are correct, I messed it up while staging changes 🤦